### PR TITLE
Apply PHP7 type hinting to the Proxy generator classes

### PR DESCRIPTION
**Pull request by lisachenko**
Improve code quality by applying type hints to Proxy classes in the source code, also clean some PHP<7.0.0 source code.

### CODE REVIEW

File: src/Proxy/TraitProxy.php [hunk 1]
```diff
@@ -36,16 +37,28 @@ class TraitProxy extends ClassProxy
...
+     * @param string $methodName Name of the method
+     *
+     * @return MethodInvocation
+     */
+    public static function getJoinPoint(
+        string $traitName,
+        string $className,
+        string $joinPointType,
+        string $methodName
+    ): MethodInvocation {
```
[the remark this answers is not in the document]
Comment (lisachenko) [reply src/Proxy/TraitProxy.php:61]: It's PSR-12 source code formatting with return types and multiline arguments

**Comment by lisachenko** [comment]
@TheCelavi would you like to review my PRs?

**Comment by TheCelavi** [comment]
On it.

File: src/Proxy/AbstractProxy.php [hunk 2]
```diff
@@ -204,9 +194,9 @@ protected function prepareArgsLine(ReflectionFunctionAbstract $functionLike)
...
      * @return string
      */
-    protected function getOverriddenFunction(ReflectionFunctionAbstract $functionLike, $body)
+    protected function getOverriddenFunction(ReflectionFunctionAbstract $functionLike, string $body): string
     {
-        $reflectionReturnType = PHP_VERSION_ID >= 70000 ? $functionLike->getReturnType() : '';
+        $reflectionReturnType = $functionLike->getReturnType();
         $modifiersLine        = '';
         if ($reflectionReturnType) {
             $nullablePrefix = $reflectionReturnType->allowsNull() ? '?' : '';
```
Comment (TheCelavi) [inline src/Proxy/AbstractProxy.php:202]: Note: this function uses built in function `join` which is alias for `implode` and as such not quite safe to use. Perhaps you should use this opportunity to replace it with `implode`?

**Comment by lisachenko** [comment]
> On it.

Superb 👍  Of course, it's simple review, but having another eyes on changes is always good for code quality

**Review by TheCelavi — APPROVED** [review]

File: src/Proxy/ClassProxy.php [hunk 3]
```diff
@@ -205,10 +198,8 @@ public function override($methodName, $body)
      * @param bool $byReference Is method should return value by reference
      * @param string $body Body of method
      * @param string $parameters Definition of parameters
-     *
-     * @return static
      */
-    public function setMethod($methodFlags, $methodName, $byReference, $body, $parameters)
+    public function setMethod(int $methodFlags, string $methodName, bool $byReference, string $body, string $parameters)
     {
```
Comment (TheCelavi) [inline src/Proxy/ClassProxy.php:203]: This method uses `join` as well instead of `implode`

File: src/Proxy/ClassProxy.php [hunk 4]
```diff
@@ -341,27 +310,21 @@ public function addTrait($trait)
...
      * @param null|string $defaultText Default value, should be string text!
-     *
-     * @return static
      */
-    public function setProperty($propFlags, $propName, $defaultText = null)
+    public function setProperty(int $propFlags, string $propName, string $defaultText = null)
     {
         $this->propertiesCode[$propName] = (
             "/**\n * Property was created automatically, do not change it manually\n */\n" . // Doc-block
             join(' ', Reflection::getModifierNames($propFlags)) . // List of modifiers for property
```
Comment (TheCelavi) [inline src/Proxy/ClassProxy.php:318]: `join` here as well.

**Review by TheCelavi — APPROVED** [review]
It seams that there are no issues here, except you may use this opportunity to replace `join` with `implode` and prevent possible BC breaks if PHP ever decides to drop support for `join`.

**Comment by lisachenko** [comment]
Ok, thank you for reviewing! 

I don't mind replacing `join` with `implode` in one single PR.

File: src/Proxy/ClassProxy.php (hunk 4 above)
Comment (lisachenko) [reply src/Proxy/ClassProxy.php:318]: Will be in a separate PR for `join` => `implode` replacement.

**Pull request merged**